js/StickyScrollBarComponent.js

```diff
@@ -0,0 +1,138 @@
+    }
+
+    /**
+     * Initialise
+     * @param {jQuery} $element - jQuery object of the element that requires a sticky scroll bar
+     */
+    init ($element) {
+        if (typeof $element === 'undefined' || !$element) {
+            throw new Error('an element to add the scroller to must be passed to StickyScrollBarComponent');
+        }
+
+        if (typeof this.$window === 'undefined' || !this.$window) {
+            throw new Error('window must be passed to StickyScrollBarComponent');
+        }
+
+        if (typeof this.$html === 'undefined' || !this.$html) {
+            throw new Error('$html must be passed to StickyScrollBarComponent');
+        }
+
+        // Set $elementWithStickyScrollBar
+        this.$elementWithStickyScrollBar = $element;
+    
+        // Add the sticky scroll bar
+        this.$stickyScrollBar.appendTo(this.$elementWithStickyScrollBar);
+
+        // Init the sticky scroll bar
+        this.$stickyScrollBar
+            .addClass('u-display-none')
+            .on('scroll', this.scrollElementWithStickyScrollBarByScrollBarWithContext);
+
+        // Call on load
+        this.updateStickyScrollBarWithContext();
+
+        // Call on scroll and window resize
+        this.$window.on('scroll resize', this.updateStickyScrollBarWithContext);
+    }
+
+    /**
+     * Set $elementWithStickyScrollBar scroll when scrolled by sticky scroll bar
+     */
+    scrollElementWithStickyScrollBarByScrollBar () {
+        this.$elementWithStickyScrollBar.scrollLeft(this.$stickyScrollBar.scrollLeft());
+    }
+
+    /**
+     * Set $stickyScrollBar scroll when $elementWithStickyScrollBar is scrolled
+     */
+    scrollStickyScrollBar () {
+        this.$stickyScrollBar.scrollLeft(this.$elementWithStickyScrollBar.scrollLeft());
+    }
+
+    /**
+     * Toggle visiblity of sticky scroll bar
+     */
+    showStickyScrollBar (option) {
+        if (option) {
+            this.$stickyScrollBar.removeClass('u-display-none');
+        } else {
+            this.$stickyScrollBar.addClass('u-display-none');
+        }
+    }
+
+    /**
+     * Update sticky scroll bar visibility, thumb position and width
+     */
+    updateStickyScrollBar () {
+        const top = this.$elementWithStickyScrollBar.offset().top;
+        const bottom = top + this.$elementWithStickyScrollBar.height();
+        const topOffset = 30;
+        let viewportBottom;
+
+        // Allow for footer
+        if (this.$html.find('.footer').css('position') === 'fixed') {
+            viewportBottom = this.$window.scrollTop() + this.$window.height() - this.$html.find('.footer').outerHeight();
+        } else {
+            viewportBottom = this.$window.scrollTop() + this.$window.height();
+        }
+
+        // Check if the $elementWithStickyScrollBar is visible but bottom is outside of viewport
+        if (top + topOffset < viewportBottom && bottom > viewportBottom) {
+
```

Comment:
empty line

Reply:
Added for readability as the line below is also a comment. Personal preference, happy to remove.